### ISSUE TITLE
Choose when to highlight more dynamically

### ISSUE THREAD

**Issue by korreman** [hadronized/kak-tree-sitter#260]
I currently get `unsupported language: <lang>` errors specifically when I first open a supported file, then open an unsupported file in a new window with `:new edit <file>`. The same error as can be achieved by changing `kts_lang` to some nonsense.

Issue seems to be that a `highlight-enable` request is sent for the new client with the language of the previous window. A correct request is sent and denied right after, but by then highlighting has already been enabled. I think the more general issue is that we decide whether to highlight or not when a window is opened, and then never again.

A more reliable behavior might be to re-decide whether to highlight a buffer whenever its `kts_lang` is changed. For this to work, a hook would need to update `kts_lang` whenever `filetype` is changed. By removing this specific hook, you could then also get dynamic control over the highlighting though `kts_lang`.

**Comment by hadronized** [comment]
Indeed, I thought I had that hook lying around already. We can simply put [this](https://github.com/hadronized/kak-tree-sitter/blob/master/kak-tree-sitter/rc/static.kak#L110) in it.

**Comment by hadronized** [comment]
I need to make a release of KTS but this is not going to be included because it’s a breaking change; I’m moving it to the next major.
